### PR TITLE
(Cont 778) -  add puppet 8 support/drop puppet 6 support

### PR DESCRIPTION
**Pull request by jordanbreen28**
This PR ensures this modules compatibility with the upcoming puppet 8 release.
This PR also drops support for puppet 6.

### CODE REVIEW

File: .fixtures.yml [hunk 1]
```diff
@@ -5,6 +5,6 @@ fixtures:
     provision: 'https://github.com/puppetlabs/provision.git'
     puppet_agent:
       repo: 'https://github.com/puppetlabs/puppetlabs-puppet_agent.git'
-      ref: v4.12.1
+      ref: v4.13.0
```
Comment (GSPatton) [inline .fixtures.yml:8]: i think we're removing the pin here altogether

File: lib/facts.rb [hunk 2]
```diff
@@ -18,16 +18,17 @@ def initialize
   # Resolves fact on the system, if a specific fact is given,
   # it will only resolve that fact,
   # otherwise it will resolve all facts
-  def resolve(fact)
+  def resolve(fact) # rubocop:disable Metrics/MethodLength
```
Comment (GSPatton) [inline lib/facts.rb:21]: instead of disabling cops in code we are leaving exceptions in rubocop_todo.yml

Comment (jordanbreen28) [reply lib/facts.rb]: @GSPatton removed :-)

**Review by GSPatton — APPROVED** [review]
all looks good to me

**Pull request merged**